engine-nextgen/src/main/kotlin/graphql/nadel/NextgenEngine.kt

```diff
@@ -129,6 +133,22 @@ class NextgenEngine @JvmOverloads constructor(
         return result
     }
 
+    private suspend fun resolveServiceAndExecuteTopLevelField(
+        field: ExecutableNormalizedField,
+        executionContext: NadelExecutionContext,
+        service: Service
+    ): ExecutionResult = if (usesDynamicServiceResolution(topLevelField = field)) {
+        val serviceOrError = serviceExecutionHooks.resolveServiceForField(services.values, field)
+
+        if (serviceOrError.error != null) {
+            newExecutionResult(error = serviceOrError.error, data = mapOf(field.resultKey to null))
```

Comment:
I think it better be `mutableMapOf`

Reply:
Hm, the type of `data` in `ExecutionResult` is `Object`. Do consumer expect it to be a mutable data structure?

Reply:
currently the `result.data` is always recursively mutable. Not 100% sure whether it's by design or by coincidence, maybe @gnawf or @andimarek can clarify.

Reply:
At this point, I don't think it actually needs to be mutable, but we should set it to mutable anyway.

The reason the data needs to be mutable is because of `NadelResultTransformer` which transforms the JSON structure (Java `Map`s and `List`s) assuming that they're mutable. We don't create new `List`s or `Map`s.